Tests/ApolloCodegenTests/AnimalKingdomAPI/AnimalKingdomASTCreationTests.swift

```diff
@@ -12,61 +12,71 @@ final class AnimalKingdomASTCreationTests: XCTestCase {
   }
 
   override func tearDown() {
     super.tearDown()
+    compilationResult = nil
+    expected = nil
   }
 
-  func test__mergedSelections_AllAnimalsQuery_RootQuery__isCorrect() {
+  func test__mergedSelections_AllAnimalsQuery_RootQuery__isCorrect() throws {
     // given
-    let operation = Self.compilationResult.operations.first { $0.name == "AllAnimalsQuery" }
-    let rootSelectionSet = ASTSelectionSet(selectionSet: operation!.selectionSet,
-                                           compilationResult: Self.compilationResult)
+    let operation = compilationResult.operations.first { $0.name == "AllAnimalsQuery" }
+    let ir = IR(compilationResult: compilationResult)
+    let rootSelectionSet = ir.build(operation: try XCTUnwrap(operation)).rootField.selectionSet!
 
-    let expected = SortedSelections(
+    expected = (
       fields: [
         .mock("allAnimals",
-              type: .nonNull(.list(.nonNull(.named(GraphQLInterfaceType.mock("Animal"))))))
-      ]
+              type: .nonNull(.list(.nonNull(.entity(GraphQLInterfaceType.mock("Animal"))))))
+      ],
+      typeCases: [],
+      fragments: []
     )
 
     // when
     let actual = rootSelectionSet.mergedSelections
 
     // then
-    expect(actual).to(shallowlyMatch(expected))
+    expect(actual).to(shallowlyMatch(self.expected))
   }
 
-  func test__mergedSelections_AllAnimalsQuery_AllAnimal__isCorrect() {
+  func test__mergedSelections_AllAnimalsQuery_AllAnimal__isCorrect() throws {
     // given
     let Interface_Animal = GraphQLInterfaceType.mock("Animal")
 
-    let operation = Self.compilationResult.operations.first { $0.name == "AllAnimalsQuery" }
-    guard case let .field(allAnimals) = operation!.selectionSet.selections[0] else { fail(); return }
-    let scope = ASTSelectionSet(selectionSet: allAnimals.selectionSet!,
-                                compilationResult: Self.compilationResult)
+    let operation = compilationResult.operations.first { $0.name == "AllAnimalsQuery" }
+    let ir = IR(compilationResult: compilationResult)
+    let rootSelectionSet = ir.build(operation: try XCTUnwrap(operation)).rootField.selectionSet!
+
+    let selectionSet = try XCTUnwrap(rootSelectionSet[field: "allAnimals"]?.selectionSet)
 
-    let expected = SortedSelections(
+    expected = (
       fields: [
         .mock("height",
-              type: .nonNull(.named(GraphQLObjectType.mock("Height")))),
+              type: .nonNull(.entity(GraphQLObjectType.mock("Height")))),
         .mock("species",
-              type: .nonNull(.named(GraphQLScalarType.string()))),
+              type: .nonNull(.scalar(GraphQLScalarType.string()))),
         .mock("skinCovering",
-              type: .named(GraphQLEnumType.skinCovering())),
+              type: .enum(GraphQLEnumType.skinCovering())),
         .mock("predators",
-              type: .nonNull(.list(.nonNull(.named(Interface_Animal))))),
+              type: .nonNull(.list(.nonNull(.entity(Interface_Animal))))),
```

Comment:
The ability to tell what these are in terms of entity vs scalar vs enum is aaawesome